desc/protoparse/parser.go

```diff
@@ -469,6 +469,12 @@ func parseProtoFiles(acc FileAccessor, filenames []string, recursive, validate b
 		if err != nil {
 			if d, ok := standardImports[name]; ok {
 				parsed[name] = &parseResult{fd: d}
+				if recursive {
```

Comment:
This is a bit of a shame to copy-pasta this whole block.

I've given a go at re-arranging the flow so that it need not be copied. What do you think? #222 

I also added a test that was able to reproduce the issue before applying the patch in `parser.go`, to verify that the issue goes away with the patch.